src/libmongoc/tests/test-mongoc-ssl.c

```diff
@@ -0,0 +1,194 @@
+         "'test_ca_file', 'tlsAllowInvalidCertificates': true, "
+         "'tlsAllowInvalidHostnames': true, 'tlsInsecure': true, "
+         "'tlsDisableCertificateRevocationCheck': true, "
+         "'tlsDisableOCSPEndpointCheck': true }",
+         NULL /* expect_error */,
+         "test_pem_file" /* pem_file */,
+         "test_pem_pwd" /* pem_pwd */,
+         "test_ca_file" /* ca_file */,
+         true /* weak_cert_validation */,
+         true /* allow_invalid_hostname */,
+         true /* disable_ocsp_endpoint_check */,
+         true /* disable_certificate_revocation_check */
+      },
+      {
+         "test options are case insentive",
+         "{'tlscertificatekeyfile': 'test_pem_file', "
+         "'tlscertificatekeyfilepassword': 'test_pem_pwd', 'tlscafile': "
+         "'test_ca_file', 'tlsallowinvalidcertificates': true, "
+         "'tlsallowinvalidhostnames': true, 'tlsinsecure': true, "
+         "'tlsdisablecertificaterevocationcheck': true, "
+         "'tlsdisableocspendpointcheck': true }",
+         NULL /* expect_error */,
+         "test_pem_file" /* pem_file */,
+         "test_pem_pwd" /* pem_pwd */,
+         "test_ca_file" /* ca_file */,
+         true /* weak_cert_validation */,
+         true /* allow_invalid_hostname */,
+         true /* disable_ocsp_endpoint_check */,
+         true /* disable_certificate_revocation_check */
+      },
+      {
+         "test no options set",
+         "{}",
+         NULL /* expect_error */,
+         NULL /* pem_file */,
+         NULL /* pem_pwd */,
+         NULL /* ca_file */,
+         false /* weak_cert_validation */,
+         false /* allow_invalid_hostname */,
+         false /* disable_ocsp_endpoint_check */,
+         false /* disable_certificate_revocation_check */
+      },
+      {
+         "test tlsInsecure overrides tlsAllowInvalidHostnames and "
+         "tlsAllowInvalidCertificates set",
+         "{'tlsInsecure': true, 'tlsAllowInvalidHostnames': false, "
+         "'tlsAllowInvalidCertificates': false}",
+         NULL /* expect_error */,
+         NULL /* pem_file */,
+         NULL /* pem_pwd */,
+         NULL /* ca_file */,
+         true /* weak_cert_validation */,
+         true /* allow_invalid_hostname */,
+         false /* disable_ocsp_endpoint_check */,
+         false /* disable_certificate_revocation_check */
+      },
+      {
+         "test unrecognized option",
+         "{'foo': true }",
+         "unexpected option: foo" /* expect_error */,
+         NULL /* pem_file */,
+         NULL /* pem_pwd */,
+         NULL /* ca_file */,
+         false /* weak_cert_validation */,
+         false /* allow_invalid_hostname */,
+         false /* disable_ocsp_endpoint_check */,
+         false /* disable_certificate_revocation_check */
+      },
+      {0}};
+   testcase_t *test;
+
+   for (test = tests; test->bson != NULL; test++) {
+      mongoc_ssl_opt_t ssl_opt = {0};
+      bson_string_t *errmsg;
```

Comment:
Since we've initialized ssl_opt, it would be ok with me to see the other variables also initialized-- but, fine as is also.

Reply:
Moved the initial assignments into the declarations.